### PR TITLE
test(heading): typescript refactor

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
- Rename the `heading.cy.js` to `heading.cy.tsx` file in `./cypress/components/heading/` folder.
- Refactor tests to Typescript.  
- Added eslint comment for 'badge.cy.tsx'
### Current behaviour

Currently Badge component is in JS not in TS.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run `npx cypress open --component` to check if the`heading.cy.tsx` file passed
- [x] Run `npx cypress run --component` to check none of the other *.cy.tsx files have regressed

### CODE REVIEW

File: cypress/components/heading/heading.cy.tsx [hunk 1]
```diff
@@ -149,13 +150,18 @@ context("Testing Heading component", () => {
     );
 
     describe("when headingType prop is provided", () => {
-      it.each(["h1", "h2", "h3", "h4", "h5"])(
+      it.each([
+        "h1" as const,
```
Comment (DlgSHi) [inline cypress/components/heading/heading.cy.tsx:154]: shall we make this `as HeadingProps["headingType"]` ?

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: cypress/components/heading/heading.cy.tsx [hunk 2]
```diff
@@ -149,13 +151,15 @@ context("Testing Heading component", () => {
     );
 
     describe("when headingType prop is provided", () => {
-      it.each(["h1", "h2", "h3", "h4", "h5"])(
+      it.each(["h1", "h2", "h3", "h4", "h5"] as Exclude<
```
Comment (DlgSHi) [inline cypress/components/heading/heading.cy.tsx:154]:
```suggestion
      it.each(["h1", "h2", "h3", "h4", "h5"] as HeadingProps["headingType"][]
```

File: cypress/components/heading/heading.cy.tsx [hunk 3]
```diff
@@ -149,13 +151,15 @@ context("Testing Heading component", () => {
...
+        HeadingProps["headingType"],
```
Comment (DlgSHi) [inline cypress/components/heading/heading.cy.tsx:155]:
```suggestion
```

File: cypress/components/heading/heading.cy.tsx [hunk 4]
```diff
@@ -149,13 +151,15 @@ context("Testing Heading component", () => {
...
+        undefined
```
Comment (DlgSHi) [inline cypress/components/heading/heading.cy.tsx:156]:
```suggestion
```

File: cypress/components/heading/heading.cy.tsx [hunk 5]
```diff
@@ -149,13 +151,15 @@ context("Testing Heading component", () => {
...
+      >[])(
```
Comment (DlgSHi) [inline cypress/components/heading/heading.cy.tsx:157]:
```suggestion
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]
Also, we need to mention in PR description about the `badge` commit

**Review by Parsium — COMMENTED** [review]
Looks good to me @divyajindel 👍🏼 Please could you add those changes to `badge.cy.tsx` as a separate commit? That way it makes it a bit clearer that the changes are separate to refactoring the `Heading` tests

**Review by Parsium — DISMISSED** [review]

**Review by Parsium — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**